sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py

```diff
@@ -196,44 +286,80 @@ def identifier_from_raw_id(raw_id: str) -> CommunicationIdentifier:
 
     :param str raw_id: A raw ID to construct the CommunicationIdentifier from.
     """
-    if raw_id.startswith('4:'):
+    if raw_id.startswith(PHONE_NUMBER_PREFIX):
         return PhoneNumberIdentifier(
-            value = raw_id[len('4:'):]
+            value=raw_id[len(PHONE_NUMBER_PREFIX):]
         )
 
     segments = raw_id.split(':', maxsplit=2)
-    if len(segments) < 3:
+    if len(segments) != 3:
+        if len(segments) == 2 and raw_id.startswith(BOT_PREFIX):
+            return MicrosoftBotIdentifier(
```

Comment:
The constructor must pass in the `raw_id` here

Reply:
Updated